assembl/static2/js/app/components/debate/thread/topPostForm.jsx

```diff
@@ -0,0 +1,217 @@
+      label: I18n.t('debate.survey.modalFooter'),
+      internalLink: true
+    };
+    if (!isUserConnected) {
+      displayModal(null, modalBody, true, null, button, true);
+    } else {
+      displayForm(true);
+    }
+  };
+
+  const emptySubject = () => {
+    return updateSubject('');
+  };
+
+  const emptyBody = () => {
+    return updateBody('');
+  };
+
+  const resetForm = () => {
+    displayForm(false);
+    emptySubject();
+    emptyBody();
+  };
+
+  const variables = {
+    ideaId: ideaId,
+    subject: subject,
+    body: body
+  };
+
+  const createTopPost = () => {
+    if (subject && body) {
+      displayAlert('success', I18n.t('loading.wait'));
+      mutate({ variables: variables })
+        .then(() => {
+          refetchIdea();
+          displayAlert('success', I18n.t('debate.thread.postSuccess'));
+          resetForm();
+        })
+        .catch((error) => {
+          displayAlert('danger', error);
+        });
+    } else if (!subject) {
+      displayAlert('warning', I18n.t('debate.thread.fillSubject'));
+    } else if (!body) {
+      displayAlert('warning', I18n.t('debate.thread.fillBody'));
+    }
+  };
+
+  const handleInputFocus = () => {
+    return redirectToLogin();
+  };
+
+  const handleSubjectChange = (e) => {
+    const maxChars = TEXT_INPUT_MAX_LENGTH;
+    const remaining = maxChars - e.target.value.length;
+    updateSubjectChars(remaining);
+    updateSubject(e.target.value);
+  };
+
+  const handleBodyChange = (e) => {
+    const maxChars = TEXT_AREA_MAX_LENGTH;
+    const remaining = maxChars - e.target.value.length;
+    updateBodyChars(remaining);
+    updateBody(e.target.value);
+  };
+
+  return (
+    <Row>
+      <Col xs={0} sm={1} md={2} />
+      <Col xs={12} sm={3} md={2} className="no-padding">
+        <div className="start-discussion-icon">
+          <span className="assembl-icon-discussion color" />
+        </div>
+        <div className="start-discussion">
+          <h3 className="dark-title-3 no-margin">
+            <Translate value="debate.thread.startDiscussion" />
+          </h3>
+        </div>
+      </Col>
+      <Col xs={12} sm={7} md={6} className="no-padding">
+        <div className="form-container">
+          <FormGroup>
+            {subject ? <div className="form-label">{I18n.t('debate.subject')}</div> : null}
+            <FormControl
+              type="text"
+              placeholder={I18n.t('debate.subject')}
+              maxLength={TEXT_INPUT_MAX_LENGTH}
+              value={subject}
+              onFocus={handleInputFocus}
+              onChange={handleSubjectChange}
+            />
+            <div className="annotation margin-s">
+              <Translate
+                value="debate.remaining_x_characters"
+                nbCharacters={subjectRemainingChars < 10000 ? subjectRemainingChars : TEXT_INPUT_MAX_LENGTH}
+              />
+            </div>
+            <div className={isFormActive ? 'margin-m' : 'hidden'}>
```

Comment:
Is there a css animation to transition from `hidden` to `margin-m`? Fade out/slide down? Don't think so by looks of this. Let's see how this looks during QA. TODO.